### DISCUSSION TITLE
ts-patch sometimes™ fails when used in a GitHub pipeline

### DISCUSSION THREAD

**Discussion by felabr** [nonara/ts-patch#164] [Q&A]
Hello,

we are experiencing an issue with `ts-patch` in our project. Sometimes the build pipelines on GitHub fail during the build step when `ts-patch` tries to delete some lock files that seem to be missing. Everything works fine locally, and it usually works fine on GitHub, too.

We haven't built anything fancy, it's just `yarn build`, wich runs `tspc --project tsconfig.build.json`, and the config file has basic options like `outDir` and `baseUrl` and enables `typescript-plugin-styled-components`.

So, I'm a bit lost trying to figure out what's going on here, and I'd really appreciate someone pointing me in right direction.

Is there maybe a way to make `ts-patch` ignore missing lock files when it just wants to delete them anyway?

Thank you

This is the error message we are getting:

```
node:internal/fs/utils:356
    throw err;
    ^

Error: ENOENT: no such file or directory, unlink '/home/runner/work/foo/foo/node_modules/.cache/ts-patch/locks/75af25658295a89bdbf9a1f2538d1d52.lock'
    at Object.unlinkSync (node:fs:1881:3)
    at withFileLock (/home/runner/work/foo/foo/node_modules/ts-patch/utils/file-utils.js:73:26)
    at readFile (/home/runner/work/foo/foo/node_modules/ts-patch/module/module-file.js:21:37)
    at getModuleFile (/home/runner/work/foo/foo/node_modules/ts-patch/module/module-file.js:79:36)
    at getTsModule (/home/runner/work/foo/foo/node_modules/ts-patch/module/ts-module.js:41:54)
    at getLiveModule (/home/runner/work/foo/foo/node_modules/ts-patch/module/get-live-module.js:19:50)
    at Object.<anonymous> (/home/runner/work/foo/foo/node_modules/ts-patch/compiler/tsc.js:11:26)
    at Module._compile (node:internal/modules/cjs/loader:1364:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
    at Module.load (node:internal/modules/cjs/loader:1203:32) {
  errno: -2,
  syscall: 'unlink',
  code: 'ENOENT',
  path: '/home/runner/work/foo/foo/node_modules/.cache/ts-patch/locks/75af25658295a89bdbf9a1f2538d1d52.lock'
}
```

**Comment by mannycarrera4** [comment]
Was there an answer to this or a follow up or a workaround?

**Reply by felabr** [reply]
Hi @mannycarrera4,

I noticed this happens when we run two builds in parallel for cjs and es, like this: `tspc --project tsconfig.build.cjs.json & tspc --project tsconfig.build.es.json`.

As a workaround we now run them one after the other, like this: `tspc --project tsconfig.build.json && tspc --project tsconfig.build.es.json` (with `&&` instead of `&`).

So I assume the names of the lock files are sometimes not unique when we do parallel builds.

I've created PR #165 that would simply ignore missing lock files that may already have been deleted.  But if the above assumption is right it would probably be better to ensure unique lock file names even in parallel builds.

**Reply by mannycarrera4** [reply]
Ah thanks for getting back to me! Yeah we had to do the same, remove `--parallel` from our build script and that seemed to have worked.

**Comment by nonara** [comment]
This should be fixed in the latest version (v3.3.0) . Let me know if not, otherwise, we can close this out

**Reply by felabr** [reply]
Hi @nonara,
we've been doing parallel builds again with v3.3.0 and have not experienced any issues yet. So yes, I'd consider this issue fixed.
Thank you
